app/services/container_dashboard_service.rb

```diff
@@ -78,4 +81,82 @@ def providers
       }
     ]
   end
+
+  def heatmaps
+    # Get latest hourly rollup for each node.
+    node_ids = @ems.container_nodes if @ems
+    metrics = MetricRollup.get_latest_hourly_rollups(ContainerNode.name, node_ids)
+
+    {
+      :nodeCpuUsage    => metrics.collect { |m|
+        avg_cpu = m.cpu_usage_rate_average.round(2)
+        {
+          :id      => m.resource_id,
+          :tooltip => "#{m.resource.name} - #{avg_cpu}%",
+          :value   => avg_cpu / 100.0 # 1% should be 0.01
+        }
+      },
+      :nodeMemoryUsage => metrics.collect { |m|
+        avg_mem = m.mem_usage_absolute_average.round(2)
+        {
+          :id      => m.resource_id,
+          :tooltip => "#{m.resource.name} - #{avg_mem}%",
+          :value   => avg_mem / 100.0 # 1% should be 0.01
+        }
+      }
+    }
+  end
+
+  def node_utilization
+    total_cores = 0
+    used_cores = 0
+    total_mem = 0
+    used_mem = 0
+
+    cpu_graph_xdata = ['dates']
+    cpu_graph_ydata = ['used']
+    mem_graph_xdata = ['dates']
+    mem_graph_ydata = ['used']
+
+    time_profile = VimPerformanceDaily.default_time_profile(:ext_options => {:tz => @controller.current_user.get_timezone})
+    metrics = VimPerformanceDaily.find_entries(:time_profile => time_profile)
+    metrics = metrics.where(:resource_type => 'ExtManagementSystem')
+    metrics = metrics.where(:resource_id => @ems.id) if @ems
+    metrics = metrics.select("sum(derived_vm_numvcpus * cpu_usage_rate_average / 100) as used_cores,
+                              sum(mem_usage_absolute_average * derived_memory_available / 100) as used_mem,
+                              sum(derived_vm_numvcpus) as total_cores,
+                              sum(derived_memory_available) as total_mem,
+                              cast(timestamp as date) as day")
+    metrics = metrics.order("day").group("day")
```

Comment:
If there is no `@ems`, then won't line 123 end up bringing back *all* EMSes including cloud and infra?